### PR TITLE
HDDS-3651 Add tests for SCMRatisRequest.

### PR DESCRIPTION
**Pull request by timmylicheng**
## What changes were proposed in this pull request?
HDDS-3651 Add tests for SCMRatisRequest.

(Please fill in changes proposed in this fix)

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-3651

(Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HDDS-XXXX. Fix a typo in YYY.)

Please replace this section with the link to the Apache JIRA)

## How was this patch tested?
UT
(Please explain how this patch was tested. Ex: unit tests, manual tests)
(If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

### CODE REVIEW

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestSCMRatisRequest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      request.encode();
+    } catch (InvalidProtocolBufferException e) {
+      // Should reach here
+    } catch (Throwable t) {
+      Assert.fail("Should not reach here.");
+    }
+  }
+
+  @Test
+  public void testDecodeWithNonProto() {
```
Comment (xiaoyuyao) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestSCMRatisRequest.java:62]: NIT: We can use the @Test annotation to simplify the test case.
 @Test(expected = InvalidProtocolBufferException.class)

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestSCMRatisRequest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    String operation = "test";
+    SCMRatisRequest request = SCMRatisRequest.of(PIPELINE, operation, args);
+    Assert.assertEquals(operation,
+        SCMRatisRequest.decode(request.encode()).getOperation());
+    Assert.assertEquals(args[0],
+        SCMRatisRequest.decode(request.encode()).getArguments()[0]);
+  }
+
+  @Test
+  public void testEncodeWithNonProto() {
```
Comment (xiaoyuyao) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestSCMRatisRequest.java:47]: Same as below for the negative case.

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestSCMRatisRequest.java (hunk 1 above)
Comment (timmylicheng) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestSCMRatisRequest.java]: Updated.

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestSCMRatisRequest.java (hunk 2 above)
Comment (timmylicheng) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestSCMRatisRequest.java]: Updated.

**Comment by ChenSammi** [comment]
+1.

**Pull request merged**